test/integration/tests/tribeRedeemer.ts

```diff
@@ -0,0 +1,228 @@
+    console.log('amountsOut[3]', amountsOut[3].toString());*/
+    expect(amountsOut[0]).to.be.at.least(ethers.constants.WeiPerEther.mul(109).div(100)); // [1.09, 1.12] stETH
+    expect(amountsOut[0]).to.be.at.most(ethers.constants.WeiPerEther.mul(112).div(100)); // [1.09, 1.12] stETH
+    expect(amountsOut[1]).to.be.at.least(ethers.constants.WeiPerEther.mul(23));
+    expect(amountsOut[1]).to.be.at.most(ethers.constants.WeiPerEther.mul(24));
+    expect(amountsOut[2]).to.be.at.least(ethers.constants.WeiPerEther.mul(330));
+    expect(amountsOut[2]).to.be.at.most(ethers.constants.WeiPerEther.mul(335));
+    expect(amountsOut[3]).to.be.at.least(ethers.constants.WeiPerEther.mul(660));
+    expect(amountsOut[3]).to.be.at.most(ethers.constants.WeiPerEther.mul(670));
+  });
+
+  it('redeem() 1,000,000 TRIBE, twice', async () => {
+    const signer = await getImpersonatedSigner(addresses.core);
+    await forceEth(signer.address);
+    const amount = ethers.constants.WeiPerEther.mul(1_000_000);
+
+    const tribeBalance0 = await contracts.tribe.balanceOf(signer.address);
+    const stethBalance0 = await contracts.steth.balanceOf(signer.address);
+    const daiBalance0 = await contracts.dai.balanceOf(signer.address);
+
+    // first redeem
+    await contracts.tribe.connect(signer).approve(contracts.tribeRedeemer.address, amount);
+    await contracts.tribeRedeemer.connect(signer).redeem(signer.address, amount);
+
+    const tribeBalance1 = await contracts.tribe.balanceOf(signer.address);
+    const stethBalance1 = await contracts.steth.balanceOf(signer.address);
+    const daiBalance1 = await contracts.dai.balanceOf(signer.address);
+
+    // second redeem
+    await contracts.tribe.connect(signer).approve(contracts.tribeRedeemer.address, amount);
+    await contracts.tribeRedeemer.connect(signer).redeem(signer.address, amount);
+
+    const tribeBalance2 = await contracts.tribe.balanceOf(signer.address);
+    const stethBalance2 = await contracts.steth.balanceOf(signer.address);
+    const daiBalance2 = await contracts.dai.balanceOf(signer.address);
+
+    // should spend the TRIBE in each call
+    expect(tribeBalance0.sub(tribeBalance1)).to.be.equal(amount);
+    expect(tribeBalance1.sub(tribeBalance2)).to.be.equal(amount);
+
+    // both redeems should give the same amounts of tokens
+    expect(stethBalance1.sub(stethBalance0)).to.be.equal(stethBalance2.sub(stethBalance1));
+    expect(daiBalance1.sub(daiBalance0)).to.be.equal(daiBalance2.sub(daiBalance1));
+
+    // sanity checks on the amounts
+    const stethPerRedeem = stethBalance1.sub(stethBalance0);
+    const daiPerRedeem = daiBalance1.sub(daiBalance0);
+    expect(stethPerRedeem).to.be.at.least(ethers.constants.WeiPerEther.mul(109));
+    expect(stethPerRedeem).to.be.at.most(ethers.constants.WeiPerEther.mul(112));
+    expect(daiPerRedeem).to.be.at.least(ethers.constants.WeiPerEther.mul(66000));
+    expect(daiPerRedeem).to.be.at.most(ethers.constants.WeiPerEther.mul(67000));
```

Comment:
would be even better to assert a difference smaller than like 1bps from the last claim imo

Reply:
10 lines above an exact equality between the 2 claims is made : 
```
    // both redeems should give the same amounts of tokens
    expect(stethBalance1.sub(stethBalance0)).to.be.equal(stethBalance2.sub(stethBalance1));
    expect(daiBalance1.sub(daiBalance0)).to.be.equal(daiBalance2.sub(daiBalance1));
```
So I think we're good